src/app/components/categories/categories.service.ts

```diff
@@ -1,21 +1,17 @@
 import {Injectable}    from '@angular/core';
 import {Http} from '@angular/http';
 import 'rxjs/add/operator/toPromise';
-
 import {Category} from './category';
 import {Config} from '../../../config';
 import {ErrorService} from '../../shared/ErrorService';
-
-
+import {Observable} from 'rxjs';
 @Injectable()
 export class CategoriesService {
   constructor(private http: Http, private errorService: ErrorService) {
   }
-
-  getCategories(): Promise<Category[]> {
+  getCategories(): Observable<Category[]> {
     return this.http.get(Config.getResourceUrl('categories'))
-      .toPromise()
-      .then(response => response.json() as Category[])
+      .map(response => response.json() as Category[])
```

Comment:
Obsłużyć wyjątek. np. sprawdzić czy response to json a dopiero pozniej go sparsowac.